sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/PartitionKeyBuilder.java

```diff
@@ -118,4 +119,26 @@ public PartitionKey build() {
         partitionKeyInternal = PartitionKeyInternal.fromObjectArray(valueArray, true);
         return new PartitionKey(partitionKeyInternal);
     }
+
+    /**
+     * Returns the PartitionKey from an array of objects that is generated by PartitionKeyInternal
+     * @param values The object array of values to construct the PartitionKey with
+     * @param strict The boolean value for if to use strict on object types
+     * @return The PartitionKey
+     */
+    public static PartitionKey fromObjectArray(Object[] values, boolean strict) {
+        return new PartitionKey(PartitionKeyInternal.fromObjectArray(values, strict));
+    }
+
+    /**
+     * Returns the PartitionKey extracted from the document by PartitionKeyHelper
+     * @param document The JsonSerializable object to get the PartitionKey value from
+     * @param partitionKeyDefinition The PartitionKeyDefinition to use to extract the PartitionKey value
+     * @return The PartitionKey
+     */
+    public static PartitionKey extractPartitionKeyFromDocument(
+        JsonSerializable document,
```

Comment:
This is not doable - we can't expose Jackson API in public surface area. You could use Map[String Object] instead - that is what we will do for custom serialization as well.

Reply:
And there is efficient conversion between ObjectNode and Map[String, Object]

Reply:
Also, why do we need both in the first place in public API. With the first overload anyone could build the extraction from document?

Reply:
Please also add test coverage for this.

Reply:
arg checks in public API also need to be added - null-checks etc. including test coverage for those

Reply:
@FabianMeiswinkel I have been thinking more about this approach and I feel I should just be using PartitionKeyBuilder.build(), however I would still like to avoid doing the below checks. Instead of what I am doing here could we add a function addValues(Object[] values) to where you don't have to add each value individually to the PartitionKeyBuilder? Or is there a concern there that we could have values that aren't one of string, double or boolean?

![image](https://github.com/Azure/azure-sdk-for-java/assets/63077034/f4a35410-993d-40ad-80a5-b7a1e30e8130)

Reply:
The original purpose of exposing extractPartitionKeyFromDocument() is to avoid the need for the code below instead we could just pass in the entity and get back the PK. I don't see how we would do this with fromObjectArray()?

![image](https://github.com/Azure/azure-sdk-for-java/assets/63077034/2235d407-f373-4ddb-8500-d1f3a625ebc2)

Reply:
You can always conver the JsonSerializable (or ObjectNode) you have anywhere to a Map<Stirng, Object>

Reply:
And it means in the public API we don't leak internal API (or any Jackson specific API like ObjectNode)

Reply:
@FabianMeiswinkel comments addressed